avm/res/sql/server/main.bicep

```diff
@@ -652,3 +693,26 @@ type auditSettingsType = {
   @description('Optional. Specifies the identifier key of the auditing storage account.')
   storageAccountResourceId: string?
 }
+
+type secretsExportConfigurationType = {
+  @description('Required. The resource ID of the key vault where to store the secrets of this module.')
+  keyVaultResourceId: string
+
+  @description('Optional. Default to the resource group where this account is. The resource group name where the key vault is.')
+  resourceGroupName: string?
+
+  @description('Optional. The name for secret to create.')
+  sqlAdminPasswordSecretName: string?
+
+  @description('Optional. The name for secret to create.')
+  appUserPasswordSecretName: string?
+
+  @description('Optional. The value for secret to create.')
+  appUserPasswordSecretValue: string?
```

Comment:
Should likely not be part of the spec. Instead it would make more sense to me to add the secret outside of the module given that it's not connected to any output of the SQL-Server, but instead an input

Reply:
After understanding, we initially thought that there would be some built-in methods where the `Sql Server` is created, such as: `ListConnectionString()`, so that we can set the `KeyVault Secrets` more conveniently than splicing it.

But the current approach is no different from our external splicing, so I agree that it is more appropriate to store these `Secrets` outside the module.😃

Reply:
Hey @zedy-wj that would indeed be the idea. I already commented on the linked issue [here](https://github.com/Azure/bicep-registry-modules/issues/2608#issuecomment-2255084879).

It makes sense to set service-specific secrets like connections strings etc. as you cannot access them from outside the module without an `existing` reference.
But adding also custom secrets to the mix (i.e., those which do not come from the service), seemingly goes a bit too far of topic.

Reply:
Hey @AlexanderSehr - Based on the current file changes, it looks like `appUserPasswordSecretValue` is part of the `connection string`. Is there a better way to get the `connection string` if we don't pass this value externally? It seems to me that currently setting `appUserPasswordSecretValue` as a `KeyVault Secret` is just for convenience (Of course, it seems more appropriate to do this outside the module at present) since it's also required in the `connection string`.😄

Reply:
Fair point. I think the below as a value defined in the module that is stored in a key vault makes sense, as it references information that is only available from inside the module, and that it also neds the appUserName + password seems like a service-specific special case to me.

```bicep
{
  name: secretsExportConfiguration.?sqlAzureConnectionStringSercretName
  value: 'Server=${server.properties.fullyQualifiedDomainName}; Database=${!empty(databases) ? databases[0].name : ''}; User=${secretsExportConfiguration.?appUserName}; Password=${secretsExportConfiguration.?appUserPasswordSecretValue}'
}
```
the wordking of the `appUserName` & `appUserPasswordSecretValue` should potentially be altered a bit to make it more obvious that they belong to the `sqlAzureConnectionString` secret, but aside from that it seems to be the best option.

The only property I'd argue against is only this configuration:
```bicep
{
  name: secretsExportConfiguration.?appUserPasswordSecretName
  value: secretsExportConfiguration.?appUserPasswordSecretValue
}
``` 
because it does not reference anything from the service but could just as well be set outside the module.

Reply:
The latest update has been pushed, and it seems that using `administratorLogin` and `administratorLoginPassword` still works well. In this way, some inputs will not be necessary.